dlk/python/dlk/templates/src/pack_input_to_qwords.cpp

```diff
@@ -24,99 +24,97 @@ limitations under the License.
+        const auto l0 = vandq_u8(v0, vone);
+        const auto l1 = vandq_u8(v1, vone);
+        const auto m0 = vshrq_n_u8(v0, 1);
+        const auto m1 = vshrq_n_u8(v1, 1);
+        const auto ml0 = vmulq_u8(l0, coeff);
+        const auto ml1 = vmulq_u8(l1, coeff);
+        const auto mm0 = vmulq_u8(m0, coeff);
+        const auto mm1 = vmulq_u8(m1, coeff);
+        const auto al0 = vpadd_u8(vget_low_u8(ml0), vget_high_u8(ml0));
+        const auto al1 = vpadd_u8(vget_low_u8(ml1), vget_high_u8(ml1));
+        const auto am0 = vpadd_u8(vget_low_u8(mm0), vget_high_u8(mm0));
+        const auto am1 = vpadd_u8(vget_low_u8(mm1), vget_high_u8(mm1));
+        const auto bl = vpadd_u8(al0, al1);
+        const auto bm = vpadd_u8(am0, am1);
+        const auto c = vpadd_u8(bl, bm);
+        vst1_u8(reinterpret_cast<uint8_t*>(output + i*bits_per_input + j*area*bits_per_input), c);
+      }
     }
     Measurement::Stop();
-    return 0;
+    return;
   }
 #endif
 
 #ifdef USE_AVX
-  constexpr std::size_t SIMD_WIDTH = 32;
-  if ((input_depth % SIMD_WIDTH) == 0) {
-    const auto blocks = len / SIMD_WIDTH;
-    for (int i = 0; i < blocks; ++i) {
-      const auto a = _mm256_loadu_si256(reinterpret_cast<__m256i*>(input + i * SIMD_WIDTH));
-      const auto l = _mm256_movemask_epi8(_mm256_slli_epi16(a, 7));
-      const auto m = _mm256_movemask_epi8(_mm256_slli_epi16(a, 6));
-      output[i*2 + 0] = QUANTIZED_PACKED(l);
-      output[i*2 + 1] = QUANTIZED_PACKED(m);
+  if (bits_per_input == 2 && input_depth % bits_per_word == 0) {
+#pragma omp parallel for
+    for (size_t i = 0; i < area; ++i) {
+      for (size_t j = 0; j < full_words_in_depth; ++j) {
+        const auto a = _mm256_loadu_si256(reinterpret_cast<__m256i*>(input + i*blocks_in_depth*bits_per_word + j*bits_per_word));
+        const auto l = _mm256_movemask_epi8(_mm256_slli_epi16(a, 7));
+        const auto m = _mm256_movemask_epi8(_mm256_slli_epi16(a, 6));
+        output[i*bits_per_input + j*area*bits_per_input + 0] = QUANTIZED_PACKED(l);
+        output[i*bits_per_input + j*area*bits_per_input + 1] = QUANTIZED_PACKED(m);
+      }
     }
     Measurement::Stop();
-    return 0;
+    return;
   }
 #endif
 
-  for (int h = 0; h < input_height; ++h)
-      for (int w = 0; w < input_width; ++w) {
-          for (int d = 0; d < full_words_in_depth; ++d) {
-              output[current_word] = QUANTIZED_PACKED(0);
-              output[current_word + 1] = QUANTIZED_PACKED(0);
-              for (int d = 0; d < bits_per_word; ++d) {
-                for(int b = 0; b < bits_per_input; ++b)
-                  output[current_word + b] = QUANTIZED_PACKED(output[current_word + b].Raw() | ((input[input_index] >> b) & 1) << d);
-                input_index++;
-              }
-              current_word += bits_per_input;
-          }
+  for (size_t i = 0; i < area; ++i) {
+    for (size_t j = 0; j < full_words_in_depth; ++j) {
+      for(size_t b = 0; b < bits_per_input; ++b) {
```

Comment:
Could you insert a space after `for` for consistency?

Reply:
Fixed.